src/axom/core/utilities/NumericArray.hpp

```diff
@@ -383,14 +381,13 @@ class NumericArray  // NOLINT
   AXOM_HOST_DEVICE
   void verifyIndex(int AXOM_DEBUG_PARAM(idx)) const
```

Comment:
Looks like the OSX build choked here, might need to remove the AXOM_DEBUG_PARAM macro or change how it plays with assert.

Reply:
It's actually a good catch!

`AXOM_DEBUG_PARAM` is guarded by `AXOM_DEBUG`, while `assert` is guarded by `NDEBUG` and the two can be independently controlled.

Reply:
We can use `AXOM_UNUSED_VAR` inside the function instead of `AXOM_DEBUG_PARAM`.

See: https://github.com/LLNL/axom/blob/c70f36f6fe0d38ba366e93e3737a606488fa0f8b/src/axom/core/Macros.hpp#L171-L212

Reply:
Yes.  Thanks.